src/pydisagg/ihme/splitter/cat_splitter.py

```diff
@@ -0,0 +1,632 @@
+    target : str
+        Column name representing the target variable to split.
+    val : str
+        Column name for the population value.
+    prefix : str, optional
+        Prefix to apply to column names when merging, by default 'cat_pop_'.
+    """
+
+    index: List[str]
+    target: str
+    val: str
+    prefix: str = "cat_pop_"
+
+    @property
+    def columns(self) -> List[str]:
+        """
+        List of all required columns in the population DataFrame.
+
+        Returns
+        -------
+        List[str]
+            List of column names.
+        """
+        return list(set(self.index + [self.target, self.val]))
+
+    @property
+    def val_fields(self) -> List[str]:
+        """
+        List of value fields (attributes).
+
+        Returns
+        -------
+        List[str]
+            List containing 'val'.
+        """
+        return ["val"]
+
+    def apply_prefix(self) -> dict:
+        """
+        Create a mapping to rename columns with the specified prefix.
+
+        Returns
+        -------
+        dict
+            Mapping from original column names to prefixed column names.
+        """
+        return {
+            self.val: f"{self.prefix}{self.val}",
+            self.target: self.target,  # Do not prefix the target column
+            **{idx: idx for idx in self.index},  # Keep index columns unchanged
+        }
+
+
+class CatSplitter(BaseModel):
+    """
+    Class for splitting categorical data based on pattern and population data.
+
+    Parameters
+    ----------
+    data : CatDataConfig
+        Configuration for the data DataFrame.
+    pattern : CatPatternConfig
+        Configuration for the pattern DataFrame.
+    population : CatPopulationConfig
+        Configuration for the population DataFrame.
+    """
+
+    data: CatDataConfig
+    pattern: CatPatternConfig
+    population: CatPopulationConfig
+
+    def model_post_init(self, __context: Any) -> None:
+        """
+        Perform extra validation after model initialization.
+
+        Raises
+        ------
+        ValueError
+            If the match criteria in the pattern or population do not match the data.
+        """
+        if not set(self.pattern.index).issubset(self.data.index):
+            raise ValueError(
+                "Match criteria in the pattern must be a subset of the data"
+            )
+        if not set(self.population.index).issubset(
+            self.data.index + self.pattern.index
+        ):
+            raise ValueError(
+                "Match criteria in the population must be a subset of the data and the pattern"
+            )
+        # Check that the 'target' column in pattern and population matches data
+        if self.pattern.target != self.data.target:
+            raise ValueError(
+                "The 'target' column in pattern must match the 'target' column in data"
+            )
+        if self.population.target != self.data.target:
```

Comment:
Population will be merged after pattern is merged, as long as `population.index` isin `data.index + pattern.index` where `pattern.index` contains `by` and `target` the merge should be successful